client/components/site-selector/index.jsx

```diff
@@ -263,7 +293,7 @@ export default React.createClass( {
 					/>
 					<div className="site-selector__sites" ref="selector">
 						{ this.renderAllSites() }
-						{ this.renderRecentSites() }
+						{ this.renderStarredSites() }
```

Comment:
You need to have more than 14 sites for us to show groups. We should remove that check for starred sites.